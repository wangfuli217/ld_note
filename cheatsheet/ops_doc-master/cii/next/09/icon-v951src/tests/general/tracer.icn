global s
procedure main()
   &trace := -1
   s := "abcdef"
   &subject := "123456"
   &pos := 4
   every tracer(1)
   every foo(&pos | &subject | &pos[1] | &subject[2:5])
   display()
   write(&line)
   write(&file)
   write(&level)
end

procedure tracer(a)
   local i
   static j
   i := 2
   j := 3
   suspend .(a | i | j | s | &subject |
      &pos | &random | &trace | &subject[3:4] | &pos[1] |
      s[3] | s[3:5] | &random[1] | &trace[1] | &subject[2:5][1])
   suspend (a | i | j | s | &subject |
      &pos | &random | &trace | &subject[3:4] | &pos[1] |
      s[3] | s[3:5] | &random[1] | &trace[1] | &subject[2:5][1])
end

procedure foo(s)
   return s
end
