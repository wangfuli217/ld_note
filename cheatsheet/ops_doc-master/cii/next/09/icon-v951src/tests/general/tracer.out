tracer.icn   :    7  | tracer(1)
tracer.icn   :   20  | tracer suspended 1
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended 2
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended 3
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "abcdef"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "123456"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended 4
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended 0
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended -16
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "3"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "4"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "c"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "cd"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "0"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "-"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   20  | tracer suspended "2"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended 1
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended 2
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended (variable = 3)
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended (variable = "abcdef")
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &subject = "123456"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &pos = 4
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &random = 0
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &trace = -47
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &subject[3] = "3"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended "4"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended "abcdef"[3] = "c"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended "abcdef"[3+:2] = "cd"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended "0"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended "-"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   23  | tracer suspended &subject[2] = "2"
tracer.icn   :    7  | tracer resumed
tracer.icn   :   26  | tracer failed
tracer.icn   :    8  | foo(4)
tracer.icn   :   29  | foo returned 4
tracer.icn   :    8  | foo("123456")
tracer.icn   :   29  | foo returned "123456"
tracer.icn   :    8  | foo("4")
tracer.icn   :   29  | foo returned "4"
tracer.icn   :    8  | foo("234")
tracer.icn   :   29  | foo returned "234"
co-expression_1(1)

main local identifiers:

global identifiers:
   display = function display
   foo = procedure foo
   main = procedure main
   s = "abcdef"
   tracer = procedure tracer
   write = function write
10
tracer.icn
1
tracer.icn   :   13  main failed
